src/EnergyPlus/DataBranchNodeConnections.hh

```diff
@@ -64,9 +65,11 @@ namespace DataBranchNodeConnections {
     struct ComponentListData
     {
         // Members
-        std::string ParentCType;    // Parent Object Type (Cannot be SPLITTER or MIXER)
-        std::string ParentCName;    // Parent Object Name
-        std::string CType;          // Component Type (Cannot be SPLITTER or MIXER)
+        // std::string ParentCType; // Parent Object Type (Cannot be SPLITTER or MIXER)
+        DataLoopNode::ConnectionObjectType ParentObjectType = DataLoopNode::ConnectionObjectType::Invalid;
+        std::string ParentCName; // Parent Object Name
+        // std::string CType;       // Component Type (Cannot be SPLITTER or MIXER)
+        DataLoopNode::ConnectionObjectType ComponentObjectType = DataLoopNode::ConnectionObjectType::Invalid;
```

Comment:
Moved `CompSets` object types from string to enum. I see some commented lines here that need to be deleted.